gum/backend-qnx/gumprocess-qnx.c

```diff
@@ -231,31 +232,55 @@ void
 gum_process_enumerate_modules (GumFoundModuleFunc func,
                                gpointer user_data)
 {
-  GumDlIteratePhdrContext ctx = { func, user_data };
+  gchar * as_path;
+  gint fd, res;
+  gboolean carry_on = TRUE;
+  procfs_mapinfo * mapinfos;
+  gint num_mapinfos;
+  procfs_debuginfo * debuginfo;
+  gint i = 0;
 
-  dl_iterate_phdr (gum_emit_dl_module, &ctx);
-}
+  as_path = g_strdup_printf ("/proc/%d/as", getpid ());
+  fd = open (as_path, O_RDONLY);
+  g_assert (fd != -1);
+  g_free (as_path);
 
-static int
-gum_emit_dl_module (const struct dl_phdr_info * info,
-                    size_t size,
-                    void * data)
-{
-  GumDlIteratePhdrContext * ctx = (GumDlIteratePhdrContext *) data;
-  gboolean carry_on;
-  GumModuleDetails details;
-  GumMemoryRange range;
+  res = devctl (fd, DCMD_PROC_PAGEDATA, 0, 0, &num_mapinfos);
+  g_assert (res != 0);
+
+  mapinfos = g_malloc (sizeof (procfs_mapinfo) * num_mapinfos);
+  debuginfo = g_malloc (sizeof (procfs_debug) + 0x100);
+
+  res = devctl (fd, DCMD_PROC_PAGEDATA, &mapinfos,
+      sizeof (procfs_mapinfo) * num_mapinfos, NULL);
+  g_assert (res != 0);
 
-  range.base_address = info->dlpi_addr;
-  /* TODO: we don't know the size of this file. */
-  range.size = 0;
+  while (carry_on && i != num_mapinfos)
+  {
+    GumRangeDetails details;
+    GumMemoryRange range;
 
-  details.name = info->dlpi_name;
-  details.range = &range;
+    range.base_address = mapinfos[i].vaddr;
+    range.size = mapinfos[i].size;
 
-  carry_on = ctx->func (&details, ctx->user_data);
+    details.range = &range;
 
-  return carry_on ? 0 : 1;
+    debuginfo.vaddr = mapinfos[i].vaddr;
+    res = devctl (fd, DCMD_PROC_MAPDEBUG, debuginfo,
+        sizeof (procfs_debuginfo) + 0x100, NULL)
+    details.name = debuginfo.path;
+
+    if (mapinfos[i].flags & PROT_EXEC && mapinfos[i].flags & MAP_ELF)
```

Comment:
We tend to lean towards the explicit end and do:
```C
(mapinfos[i].flags & PROT_EXEC) != 0
```
Same for pointers, where we do `foo != NULL` instead of `foo`. The only cases where we do just `foo` is for gboolean/bool values.